modules/apps/object/object-storage-sugarcrm/src/main/java/com/liferay/object/storage/sugarcrm/internal/rest/manager/v1_0/SugarCRMObjectEntryManagerImpl.java

```diff
@@ -0,0 +1,570 @@
+
+		checkPortletResourcePermission(
+			ActionKeys.DELETE, objectDefinition, scopeKey,
+			dtoConverterContext.getUser());
+
+		_sugarCRMHttp.delete(
+			companyId, getGroupId(objectDefinition, scopeKey),
+			objectDefinition.getExternalReferenceCode() +
+				StringPool.FORWARD_SLASH + externalReferenceCode);
+	}
+
+	@Override
+	public Page<ObjectEntry> getObjectEntries(
+			long companyId, ObjectDefinition objectDefinition, String scopeKey,
+			Aggregation aggregation, DTOConverterContext dtoConverterContext,
+			String filterString, Pagination pagination, String search,
+			Sort[] sorts)
+		throws Exception {
+
+		checkPortletResourcePermission(
+			ActionKeys.VIEW, objectDefinition, scopeKey,
+			dtoConverterContext.getUser());
+
+		return _getObjectEntries(
+			companyId, objectDefinition, scopeKey, dtoConverterContext,
+			filterString, pagination, sorts);
+	}
+
+	@Override
+	public ObjectEntry getObjectEntry(
+			long companyId, DTOConverterContext dtoConverterContext,
+			String externalReferenceCode, ObjectDefinition objectDefinition,
+			String scopeKey)
+		throws Exception {
+
+		checkPortletResourcePermission(
+			ActionKeys.VIEW, objectDefinition, scopeKey,
+			dtoConverterContext.getUser());
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return null;
+		}
+
+		return toObjectEntry(
+			companyId, _getDateFormat(), _defaultObjectFieldNames,
+			dtoConverterContext,
+			_sugarCRMHttp.get(
+				companyId, getGroupId(objectDefinition, scopeKey),
+				StringBundler.concat(
+					_getObjectLocation(objectDefinition),
+					StringPool.FORWARD_SLASH, externalReferenceCode),
+				null),
+			objectDefinition);
+	}
+
+	@Override
+	public String getStorageLabel(Locale locale) {
+		return language.get(
+			locale, ObjectDefinitionConstants.STORAGE_TYPE_SUGARCRM);
+	}
+
+	@Override
+	public String getStorageType() {
+		return ObjectDefinitionConstants.STORAGE_TYPE_SUGARCRM;
+	}
+
+	@Override
+	public ObjectEntry updateObjectEntry(
+			long companyId, DTOConverterContext dtoConverterContext,
+			String externalReferenceCode, ObjectDefinition objectDefinition,
+			ObjectEntry objectEntry, String scopeKey)
+		throws Exception {
+
+		checkPortletResourcePermission(
+			ActionKeys.UPDATE, objectDefinition, scopeKey,
+			dtoConverterContext.getUser());
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return null;
+		}
+
+		JSONObject responseJSONObject = _sugarCRMHttp.put(
+			objectDefinition.getCompanyId(),
+			getGroupId(objectDefinition, scopeKey),
+			StringBundler.concat(
+				_getObjectLocation(objectDefinition), StringPool.FORWARD_SLASH,
+				externalReferenceCode),
+			toJSONObject(
+				dtoConverterContext, objectDefinition, objectEntry,
+				_getTriConsumer(objectDefinition)));
+
+		return toObjectEntry(
+			objectDefinition.getCompanyId(), _getDateFormat(),
+			_defaultObjectFieldNames, dtoConverterContext, responseJSONObject,
+			objectDefinition);
+	}
+
+	private void _appendFilter(StringBuilder sb, String filterString) {
```

Comment:
Hey @mauricesepe , you implemented the filter feature following the SugarCRM standard, but this should work following the Liferay (Odata) standard and then translating it to the SugarCRM standard. Please follow the pattern used in Salesforce implementation (see classes [SalesforceFilterFactoryImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/filter/factory/SalesforceFilterFactoryImpl.java) and [SOSQLExpressionVisitorImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java)). Thanks!